lib/airbrake/passenger.rb

```diff
@@ -0,0 +1,62 @@
+require 'set'
+require 'pathname'
+
+module Airbrake
+  # Integration for Phusion Passenger
+  module Passenger
+    IntegrationException = Class.new(::Exception)
+    class << self
+      def version
+        unless defined?(::PhusionPassenger)
+          raise IntegrationException, "PhusionPassenger not detected"
+        end
+        @version = Gem::Version.new(::PhusionPassenger::VERSION_STRING)
+      end
+
+      def version?(req)
+        reqs = req.split(',').map(&:strip).compact
+        Gem::Requirement.new(*reqs).satisfied_by?(version)
+      end
+
+      def integrations
+        return @integrations if defined?(@integrations)
+        @integrations ||= Set.new
+        dir = Pathname.new(__FILE__).dirname.join('passenger')
+        dir.each_child do |child|
+          next unless child.file? && child.readable?
+          next unless child.fnmatch?('*.rb')
+          require "airbrake/passenger/#{child.basename '.rb'}"
```

Comment:
I'm planning to do some majoring refactoring to this. I agree that this method is a bit unwieldy. I think it's use will be more clear when its more concise. The idea is to iterate through the `#iterations` to find one with version constraints that match the version of `passenger` in use, and then to use that integration.